# Analyzing CIA Factbook Data Using SQL

In this project, we'll work with data from the [CIA World Factbook](https://www.cia.gov/the-world-factbook/), a compendium of statistics about all of the countries on Earth. The Factbook contains demographic information like the following:

* `population` — the global population.

* `population_growth` — the annual population growth rate, as a percentage.

* `area` — the total land and water area.

![](https://i.ibb.co/5kwfH4f/Web-capture-2-7-2021-235452-www-cia-gov.jpg)

## Connecting Database

In [1]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

'Connected: None@factbook.db'

In [2]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type = 'table';

Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

In [3]:
%%sql
SELECT *
  FROM facts
 LIMIT 5;

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

Here are the descriptions for some of the columns:

* **name** — the name of the country.


* **area** — the country's total area (both land and water).


* **area_land** — the country's land area in square kilometers.


* **area_water** — the country's waterarea in square kilometers.


* **population** — the country's population.


* **population_growth** — the country's population growth as a percentage.


* **birth_rate** — the country's birth rate, or the number of births per year per 1,000 people.


* **death_rate** — the country's death rate, or the number of death per year per 1,000 people.


* **migration_rate** - the country's migration rate, or change in population due to migration per 1,000 people.

## Summary Statistics

In [4]:
%%sql

SELECT MIN(population)
      ,MAX(population)
      ,MIN(population_growth)
      ,MAX(population_growth)
    
 FROM facts;

Done.


[(0, 7256490011, 0.0, 4.02)]

In [5]:
%%sql

SELECT name
  FROM facts
 WHERE population = 0;

Done.


[('Antarctica',)]

In [6]:
%%sql

SELECT name
  FROM facts
 WHERE population = 7256490011;

Done.


[('World',)]

> * **Antartica** has minimum population i.e, 0


> * **World**  population is 7256490011

In [7]:
%%sql

SELECT MIN(population)
      ,MAX(population)
      ,MIN(population_growth)
      ,MAX(population_growth)
  
  FROM facts
  WHERE name != 'World';


Done.


[(0, 1367485388, 0.0, 4.02)]

## Average Population & Area

In [8]:
%%sql

SELECT AVG(population),AVG(area)
  FROM facts;

Done.


[(62094928.32231405, 555093.546184739)]

In [9]:
%%sql

SELECT name
  FROM facts
 WHERE (population > (SELECT AVG(population)
                       FROM facts))
          &
        (area < (SELECT AVG(area)
                  FROM facts));

Done.


[('Bangladesh',),
 ('Germany',),
 ('Japan',),
 ('Philippines',),
 ('Thailand',),
 ('United Kingdom',),
 ('Vietnam',)]

## Which country has the most people? Which country has the highest growth rate?

In [10]:
%%sql

SELECT name, 
       MAX(population)

  FROM facts
  WHERE name != 'World';

Done.


[('China', 1367485388)]

In [11]:
%%sql

SELECT name, 
       MAX(population_growth)

  FROM facts
  WHERE name != 'World';

Done.


[('South Sudan', 4.02)]

> * **China** has maximum population which is 136.74 Cr .
>
>
> * **South Sudan** has maximum population growth rate which is 4.02% .

## Which countries have the highest ratios of water to land? Which countries have more water than land?

In [16]:
%%sql

SELECT name, 
       CAST(area_water AS FLOAT)/area_land AS water_land_ratio
  FROM facts
 WHERE water_land_ratio > 1;

Done.


[('British Indian Ocean Territory', 905.6666666666666),
 ('Virgin Islands', 4.520231213872832)]

>* There are two country with more water area than land.
>
>
>* **British Indian Ocean Territory** has highest water-land area ratio.
>
>
>* **Virgin Islands** is other country with more than 1 water-land ratio.

## Which countries will add the most people to their populations next year?

In [20]:
%%sql

SELECT name, 
      (population * (population_growth/100)) pop_incr
  
  FROM facts
 WHERE name != 'World'
 ORDER BY pop_incr DESC
  LIMIT 1;

Done.


[('India', 15270686.124799998)]

> * **India** will add most people to population next year i.e, 1.53 Cr

## Which countries have a higher death rate than birth rate?

In [29]:
%%sql

SELECT name,
       birth_rate,
       death_rate,
       death_rate - birth_rate AS diff
        
  FROM facts
 WHERE death_rate > birth_rate
 ORDER BY death_rate DESC;

Done.


[('Ukraine', 10.72, 14.46, 3.74),
 ('Bulgaria', 8.92, 14.44, 5.52),
 ('Latvia', 10.0, 14.31, 4.3100000000000005),
 ('Lithuania', 10.1, 14.27, 4.17),
 ('Russia', 11.6, 13.69, 2.09),
 ('Serbia', 9.08, 13.66, 4.58),
 ('Belarus', 10.7, 13.36, 2.66),
 ('Hungary', 9.16, 12.73, 3.5700000000000003),
 ('Moldova', 12.0, 12.59, 0.5899999999999999),
 ('Estonia', 10.51, 12.4, 1.8900000000000006),
 ('Croatia', 9.45, 12.18, 2.7300000000000004),
 ('Romania', 9.14, 11.9, 2.76),
 ('Germany', 8.47, 11.42, 2.9499999999999993),
 ('Slovenia', 8.42, 11.37, 2.9499999999999993),
 ('Greece', 8.66, 11.09, 2.4299999999999997),
 ('Portugal', 9.27, 11.02, 1.75),
 ('Czech Republic', 9.63, 10.34, 0.7099999999999991),
 ('Italy', 8.74, 10.19, 1.4499999999999993),
 ('Poland', 9.74, 10.19, 0.4499999999999993),
 ('Bosnia and Herzegovina', 8.87, 9.75, 0.8800000000000008),
 ('Saint Pierre and Miquelon', 7.42, 9.72, 2.3000000000000007),
 ('Japan', 7.93, 9.51, 1.58),
 ('Austria', 9.41, 9.42, 0.009999999999999787),
 ('Monaco', 6.65, 9.24, 2.59)]

In [28]:
%%sql

SELECT COUNT(name)     
  FROM facts
 WHERE death_rate > birth_rate;

Done.


[(24,)]

> * Total 24 countries have higher death rate than birth rate.
>
>
> * **Ukraine** has greatest death rate in the list i.e., 14.46%.
>
>
>* **Bulgaria** has greatest decrease in net population i.e, 5.52%
>
>
> * Out of 24 countries, only two countries are not from Europe i.e, **Saint Pierre and Miquelon** & **Japan** .
>
>
> Rest 22 countrie are from **EUROPE**. This shows that there is decrese in population in European countries which is a food for thought!

## Which countries have the highest population/area ratio, and how does it compare to list we found in the previous screen?

In [31]:
%%sql

SELECT name,
       population/area AS popl_area_ratio
  
  FROM facts
 ORDER BY popl_area_ratio DESC
  LIMIT 5;


Done.


[('Macau', 21168),
 ('Monaco', 15267),
 ('Singapore', 8141),
 ('Hong Kong', 6445),
 ('Gaza Strip', 5191)]

> *  Among top 5 country with highest population area ration there is only one    country which has higher death rate than birth rate i.e, **Monaco** 

## Further Exploring Population Statistics

In [36]:
%%sql

SELECT name,
       population
 
  FROM facts
  WHERE population != 'None'
 ORDER BY population
 LIMIT 5;

Done.


[('Antarctica', 0),
 ('Pitcairn Islands', 48),
 ('Cocos (Keeling) Islands', 596),
 ('Holy See (Vatican City)', 842),
 ('Niue', 1190)]

In [46]:
%%sql

SELECT name,
       population,
       CAST(population AS FLOAT)/area_land AS popl_land_ratio
 
  FROM facts
  WHERE (population != 'None')
    AND (name != 'World' and name != 'European Union')
  ORDER BY population DESC
 LIMIT 5;

Done.


[('China', 1367485388, 146.62505594328366),
 ('India', 1251695584, 420.9937208919838),
 ('United States', 321368864, 35.07640870965904),
 ('Indonesia', 255993674, 141.31047395931373),
 ('Brazil', 204259812, 24.43842912418313)]

> * **Antartica** has least population which is **0** . It amy be due to fact that it has harsh cold waether condition throughout the year.
>
>
> * Ignoring Antartica, **Pitcairn Islands** is the least populated entity with strength of **48**. 
>
>
> * **China** is most populated country i.e, **136.74 Cr**.
>
>
>* **India** is second most populated country, but has highest population density among top 5 populous countries which is a matter of concern.


## Exporing Migration Statistics

In [54]:
%%sql

SELECT name,migration_rate
  FROM facts
  WHERE migration_rate > 0
  ORDER BY migration_rate DESC;

Done.


[('Qatar', 22.39),
 ('American Samoa', 21.13),
 ('Micronesia, Federated States of', 20.93),
 ('Syria', 19.79),
 ('Tonga', 17.84),
 ('British Virgin Islands', 17.28),
 ('Luxembourg', 17.16),
 ('Cayman Islands', 14.4),
 ('Singapore', 14.05),
 ('Nauru', 13.63),
 ('Jordan', 13.24),
 ('Bahrain', 13.09),
 ('Maldives', 12.68),
 ('United Arab Emirates', 12.36),
 ('Anguilla', 12.18),
 ('South Sudan', 11.47),
 ('Turks and Caicos Islands', 9.94),
 ('Samoa', 9.78),
 ('Moldova', 9.67),
 ('Cyprus', 9.48),
 ('Saint Vincent and the Grenadines', 9.17),
 ('Aruba', 8.92),
 ('Sao Tome and Principe', 8.63),
 ('Somalia', 8.49),
 ('Saint Pierre and Miquelon', 8.49),
 ('Spain', 8.31),
 ('El Salvador', 8.28),
 ('Puerto Rico', 8.15),
 ('Guyana', 8.06),
 ('San Marino', 8.03),
 ('Libya', 7.8),
 ('Virgin Islands', 7.67),
 ('Lesotho', 7.36),
 ('Norway', 7.25),
 ('Northern Mariana Islands', 7.16),
 ('Tuvalu', 6.81),
 ('Fiji', 6.75),
 ('Sint Maarten', 6.63),
 ('Isle of Man', 6.56),
 ('Guam', 6.34),
 ('Lithuania', 6.27),
 ('Latvia', 6.26),
 ('Trinidad and Tobago', 6.25),
 ('Djibouti', 6.06),
 ('Greenland', 5.98),
 ('Congo, Republic of the', 5.9),
 ('Belgium', 5.87),
 ('Armenia', 5.8),
 ('Canada', 5.66),
 ('Australia', 5.65),
 ('Austria', 5.56),
 ('Sweden', 5.42),
 ('Dominica', 5.38),
 ('Kyrgyzstan', 5.22),
 ('Liechtenstein', 5.08),
 ('Wallis and Futuna', 5.06),
 ('Marshall Islands', 4.83),
 ('Switzerland', 4.74),
 ('Jamaica', 4.66),
 ('Botswana', 4.56),
 ('Iceland', 4.43),
 ('Sudan', 4.29),
 ('Italy', 4.1),
 ('Ireland', 4.09),
 ('New Caledonia', 4.01),
 ('Nepal', 3.86),
 ('Timor-Leste', 3.86),
 ('United States', 3.86),
 ('Monaco', 3.83),
 ('Jersey', 3.76),
 ('Cuba', 3.66),
 ('Estonia', 3.6),
 ('Chad', 3.45),
 ('Macau', 3.37),
 ('Morocco', 3.36),
 ('Albania', 3.3),
 ('Gibraltar', 3.28),
 ('Grenada', 3.13),
 ('Finland', 3.1),
 ('Nicaragua', 3.0),
 ('Saint Lucia', 2.93),
 ('Kiribati', 2.87),
 ('Sierra Leone', 2.77),
 ('Haiti', 2.76),
 ('Georgia', 2.7),
 ('Portugal', 2.67),
 ('United Kingdom', 2.54),
 ('Comoros', 2.53),
 ('Peru', 2.53),
 ('European Union', 2.5),
 ('Brunei', 2.43),
 ('South Africa', 2.42),
 ('Uzbekistan', 2.37),
 ('Czech Republic', 2.33),
 ('Greece', 2.32),
 ('Guernsey', 2.32),
 ('Mali', 2.26),
 ('Ukraine', 2.25),
 ('Israel', 2.24),
 ('Antigua and Barbuda', 2.21),
 ('New Zealand', 2.21),
 ('Denmark', 2.2),
 ('Turkey', 2.16),
 ('Gambia, The', 2.12),
 ('Philippines', 2.09),
 ('Gabon', 2.03),
 ('Ghana', 2.02),
 ('Malta', 1.98),
 ('Mozambique', 1.98),
 ('Guatemala', 1.97),
 ('Netherlands', 1.95),
 ('Dominican Republic', 1.91),
 ('Bermuda', 1.88),
 ('Turkmenistan', 1.84),
 ('Solomon Islands', 1.75),
 ('Tunisia', 1.73),
 ('Russia', 1.69),
 ('Mexico', 1.68),
 ('Hong Kong', 1.68),
 ('Iraq', 1.62),
 ('Senegal', 1.59),
 ('Kuwait', 1.58),
 ('Pakistan', 1.54),
 ('Afghanistan', 1.51),
 ('Vanuatu', 1.47),
 ('Curacao', 1.4),
 ('Croatia', 1.39),
 ('Sri Lanka', 1.35),
 ('Hungary', 1.33),
 ('Germany', 1.24),
 ('Saint Kitts and Nevis', 1.21),
 ('Honduras', 1.16),
 ('Indonesia', 1.16),
 ('Tajikistan', 1.15),
 ('Lebanon', 1.1),
 ('France', 1.09),
 ('Laos', 1.09),
 ('Seychelles', 1.0),
 ('Yemen', 1.0),
 ('Algeria', 0.92),
 ('Uruguay', 0.9),
 ('Taiwan', 0.89),
 ('Rwanda', 0.85),
 ('Mongolia', 0.84),
 ('French Polynesia', 0.84),
 ('Costa Rica', 0.83),
 ('Mauritania', 0.83),
 ('Uganda', 0.74),
 ('Palau', 0.71),
 ('Belarus', 0.7),
 ('Zambia', 0.68),
 ('Colombia', 0.64),
 ('Cabo Verde', 0.63),
 ('Bolivia', 0.62),
 ('Niger', 0.56),
 ('Suriname', 0.56),
 ('Saudi Arabia', 0.55),
 ('Tanzania', 0.54),
 ('Macedonia', 0.48),
 ('Angola', 0.46),
 ('Bangladesh', 0.46),
 ('Poland', 0.46),
 ('China', 0.44),
 ('Oman', 0.43),
 ('Kazakhstan', 0.41),
 ('Bosnia and Herzegovina', 0.38),
 ('Slovenia', 0.37),
 ('Chile', 0.34),
 ('Malaysia', 0.33),
 ('Cambodia', 0.32),
 ('Barbados', 0.3),
 ('Vietnam', 0.3),
 ('Bulgaria', 0.29),
 ('Burma', 0.28),
 ('Panama', 0.28),
 ('Congo, Democratic Republic of the', 0.27),
 ('Romania', 0.24),
 ('Ethiopia', 0.22),
 ('Kenya', 0.22),
 ('Nigeria', 0.22),
 ('Egyp

In [59]:
%%sql

SELECT count(name)
  FROM facts
  WHERE migration_rate > 0
    AND (name != 'World' and name != 'European Union')
  ORDER BY migration_rate DESC;

Done.


[(187,)]

In [56]:
%%sql

SELECT name,migration_rate
  FROM facts
  WHERE migration_rate < 0
  ORDER BY migration_rate;

Done.


[]

In [61]:
%%sql

SELECT name,migration_rate
 FROM facts;

Done.


[('Afghanistan', 1.51),
 ('Albania', 3.3),
 ('Algeria', 0.92),
 ('Andorra', 0.0),
 ('Angola', 0.46),
 ('Antigua and Barbuda', 2.21),
 ('Argentina', 0.0),
 ('Armenia', 5.8),
 ('Australia', 5.65),
 ('Austria', 5.56),
 ('Azerbaijan', 0.0),
 ('Bahamas, The', 0.0),
 ('Bahrain', 13.09),
 ('Bangladesh', 0.46),
 ('Barbados', 0.3),
 ('Belarus', 0.7),
 ('Belgium', 5.87),
 ('Belize', 0.0),
 ('Benin', 0.0),
 ('Bhutan', 0.0),
 ('Bolivia', 0.62),
 ('Bosnia and Herzegovina', 0.38),
 ('Botswana', 4.56),
 ('Brazil', 0.14),
 ('Brunei', 2.43),
 ('Bulgaria', 0.29),
 ('Burkina Faso', 0.0),
 ('Burma', 0.28),
 ('Burundi', 0.0),
 ('Cambodia', 0.32),
 ('Cameroon', 0.15),
 ('Canada', 5.66),
 ('Cabo Verde', 0.63),
 ('Central African Republic', 0.0),
 ('Chad', 3.45),
 ('Chile', 0.34),
 ('China', 0.44),
 ('Colombia', 0.64),
 ('Comoros', 2.53),
 ('Congo, Democratic Republic of the', 0.27),
 ('Congo, Republic of the', 5.9),
 ('Costa Rica', 0.83),
 ("Cote d'Ivoire", 0.0),
 ('Croatia', 1.39),
 ('Cuba', 3.66),
 ('Cyprus', 9.48),
 ('Czech Republic', 2.33),
 ('Denmark', 2.2),
 ('Djibouti', 6.06),
 ('Dominica', 5.38),
 ('Dominican Republic', 1.91),
 ('Ecuador', 0.0),
 ('Egypt', 0.19),
 ('El Salvador', 8.28),
 ('Equatorial Guinea', 0.0),
 ('Eritrea', 0.0),
 ('Estonia', 3.6),
 ('Ethiopia', 0.22),
 ('Fiji', 6.75),
 ('Finland', 3.1),
 ('France', 1.09),
 ('Gabon', 2.03),
 ('Gambia, The', 2.12),
 ('Georgia', 2.7),
 ('Germany', 1.24),
 ('Ghana', 2.02),
 ('Greece', 2.32),
 ('Grenada', 3.13),
 ('Guatemala', 1.97),
 ('Guinea', 0.0),
 ('Guinea-Bissau', 0.0),
 ('Guyana', 8.06),
 ('Haiti', 2.76),
 ('Honduras', 1.16),
 ('Hungary', 1.33),
 ('Iceland', 4.43),
 ('India', 0.04),
 ('Indonesia', 1.16),
 ('Iran', 0.07),
 ('Iraq', 1.62),
 ('Ireland', 4.09),
 ('Israel', 2.24),
 ('Italy', 4.1),
 ('Jamaica', 4.66),
 ('Japan', 0.0),
 ('Jordan', 13.24),
 ('Kazakhstan', 0.41),
 ('Kenya', 0.22),
 ('Kiribati', 2.87),
 ('Korea, North', 0.04),
 ('Korea, South', 0.0),
 ('Kosovo', None),
 ('Kuwait', 1.58),
 ('Kyrgyzstan', 5.22),
 ('Laos', 1.09),
 ('Latvia', 6.26),
 ('Lebanon', 1.1),
 ('Lesotho', 7.36),
 ('Liberia', 0.0),
 ('Libya', 7.8),
 ('Liechtenstein', 5.08),
 ('Lithuania', 6.27),
 ('Luxembourg', 17.16),
 ('Macedonia', 0.48),
 ('Madagascar', 0.0),
 ('Malawi', 0.0),
 ('Malaysia', 0.33),
 ('Maldives', 12.68),
 ('Mali', 2.26),
 ('Malta', 1.98),
 ('Marshall Islands', 4.83),
 ('Mauritania', 0.83),
 ('Mauritius', 0.0),
 ('Mexico', 1.68),
 ('Micronesia, Federated States of', 20.93),
 ('Moldova', 9.67),
 ('Monaco', 3.83),
 ('Mongolia', 0.84),
 ('Montenegro', None),
 ('Morocco', 3.36),
 ('Mozambique', 1.98),
 ('Namibia', 0.0),
 ('Nauru', 13.63),
 ('Nepal', 3.86),
 ('Netherlands', 1.95),
 ('New Zealand', 2.21),
 ('Nicaragua', 3.0),
 ('Niger', 0.56),
 ('Nigeria', 0.22),
 ('Norway', 7.25),
 ('Oman', 0.43),
 ('Pakistan', 1.54),
 ('Palau', 0.71),
 ('Panama', 0.28),
 ('Papua New Guinea', 0.0),
 ('Paraguay', 0.07),
 ('Peru', 2.53),
 ('Philippines', 2.09),
 ('Poland', 0.46),
 ('Portugal', 2.67),
 ('Qatar', 22.39),
 ('Romania', 0.24),
 ('Russia', 1.69),
 ('Rwanda', 0.85),
 ('Saint Kitts and Nevis', 1.21),
 ('Saint Lucia', 2.93),
 ('Saint Vincent and the Grenadines', 9.17),
 ('Samoa', 9.78),
 ('San Marino', 8.03),
 ('Sao Tome and Principe', 8.63),
 ('Saudi Arabia', 0.55),
 ('Senegal', 1.59),
 ('Serbia', 0.0),
 ('Seychelles', 1.0),
 ('Sierra Leone', 2.77),
 ('Singapore', 14.05),
 ('Slovakia', 0.04),
 ('Slovenia', 0.37),
 ('Solomon Islands', 1.75),
 ('Somalia', 8.49),
 ('South Africa', 2.42),
 ('South Sudan', 11.47),
 ('Spain', 8.31),
 ('Sri Lanka', 1.35),
 ('Sudan', 4.29),
 ('Suriname', 0.56),
 ('Swaziland', 0.0),
 ('Sweden', 5.42),
 ('Switzerland', 4.74),
 ('Syria', 19.79),
 ('Tajikistan', 1.15),
 ('Tanzania', 0.54),
 ('Thailand', 0.0),
 ('Timor-Leste', 3.86),
 ('Togo', 0.0),
 ('Tonga', 17.84),
 ('Trinidad and Tobago', 6.25),
 ('Tunisia', 1.73),
 ('Turkey', 2.16),
 ('Turkmenistan', 1.84),
 ('Tuvalu', 6.81),
 ('Uganda', 0.74),
 ('Ukraine', 2.25),
 ('United Arab Emirates', 12.36),
 ('United Kingdom', 2.54),
 ('Un

>*  **migration_rate** column has only non-negative values. So, this is ambiguous as many countries have negative migration rate in reality.
>
>
>
>*  **Quatar** has highest migration rate which is 	**22.39%** .